src/visibility.js

```diff
@@ -0,0 +1,170 @@
+    }
+    const options = {
+      threshold: this.config.threshold
+    };
+
+    this.observer = new window.IntersectionObserver(this._handleVisibilityChange.bind(this), options);
+    this.observer.observe(this.appTargetContainer);
+  }
+
+  _initFloating() {
+    const defaultFloatingConfig = {
+      floating: {
+        position: 'bottom-right',
+        size: {
+          height: '225px',
+          width: '400px'
+        }
+      }
+    };
+    Utils.Object.mergeDeep(this.config, defaultFloatingConfig, Utils.Object.copyDeep(this.config));
+    this.floatingPoster = document.createElement('div');
+    this.floatingPoster.className = FLOATING_POSTER_CLASS;
+    this.floatingContainer = document.createElement('div');
+    this.floatingContainer.className = FLOATING_CONTAINER_CLASS;
+
+    // local const is defined to override flow null reference error
+    const appTargetContainer = (this.appTargetContainer = document.getElementById(this.player.config.targetId));
+    if (appTargetContainer) {
+      this.floatingContainer.innerHTML = appTargetContainer.innerHTML;
+      appTargetContainer.innerHTML = '';
+      appTargetContainer.appendChild(this.floatingPoster);
+      appTargetContainer.appendChild(this.floatingContainer);
+    }
+    this.config.floating.position.split('-').forEach(side => {
+      Utils.Dom.addClassName(this.floatingContainer, `${FLOATING_ACTIVE_CLASS}-${side}`);
+    });
+  }
+
+  _dismissed() {
+    this.dismissed = true;
+    this.player.pause();
+    this._stopFloating();
+  }
+
+  _stopFloating() {
+    Utils.Dom.removeClassName(this.floatingContainer, FLOATING_ACTIVE_CLASS);
+    Utils.Dom.removeAttribute(this.floatingContainer, 'style');
+  }
+
+  _startFloating() {
+    Utils.Dom.addClassName(this.floatingContainer, FLOATING_ACTIVE_CLASS);
+    Utils.Dom.setStyle(this.floatingContainer, 'height', this.config.floating.size.height);
+    Utils.Dom.setStyle(this.floatingContainer, 'width', this.config.floating.size.width);
+  }
+
+  _handleVisibilityChange(entries: Array<window.IntersectionObserverEntry>) {
+    const playerIsOutOfVisibility = entries[0].intersectionRatio < this.config.threshold;
+    if (this.config.floating && this.everStartedPlaying && !this.dismissed) {
+      this._handleFloatingChange(playerIsOutOfVisibility);
+    }
+  }
+
+  _handleFloatingChange(playerIsOutOfVisibility: boolean) {
+    if (playerIsOutOfVisibility) {
+      this._startFloating();
+    } else {
+      this._stopFloating();
+    }
+  }
+  /**
+   * _addBindings
+   * @private
+   * @returns {void}
+   */
+  _addBindings(): void {
+    this.eventManager.listen(this.player, this.player.Event.FIRST_PLAYING, () => {
+      this.everStartedPlaying = true;
+      Utils.Dom.setStyle(this.floatingPoster, 'background-image', `url("${this.player.config.sources.poster}")`);
+    });
+  }
+
+  /**
+   * Destroys the plugin.
+   * @override
+   * @public
+   * @returns {void}
+   */
+  destroy(): void {
+    this._clean();
```

Comment:
not defined!

Reply:
also, clean the observer here other wise it will keep working after we destroy the player